src/test/end.t.sol

```diff
@@ -483,6 +504,126 @@ contract EndTest is DSTest {
         assertEq(balanceOf("gold", address(gold.gemA)), 0);
     }
 
+    // -- Scenario where there is one collateralised CDP
+    // -- undergoing auction at the time of cage
+    function test_cage_snip() public {
+        Ilk memory gold = init_collateral("gold");
+
+        Usr ali = new Usr(vat, end);
+
+        // Make a CDP:
+        address urn1 = address(ali);
+        gold.gemA.join(urn1, 10 ether);
+        ali.frob("gold", urn1, urn1, urn1, 10 ether, 15 ether);
+        (uint ink1, uint art1) = vat.urns("gold", urn1); // CDP before liquidation
+        (, uint rate,,,) = vat.ilks("gold");
+        assertEq(vat.gem("gold", urn1), 0);
+        assertEq(rate, ray(1 ether));
```

Comment:
I get very nervous when we only test with `rate == RAY`...it makes it very easy for bugs stemming from an incorrect use of `rate` to slip by. Do you mind either changing this test to use a non-zero `rate`, or else adding another test that does so?